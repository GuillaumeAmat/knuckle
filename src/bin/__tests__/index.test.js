const packageJson = require('../../../package.json');
const binScriptPath = require('../../helpers/tests/binScriptPath');
const cmd = require('../../helpers/tests/cmd');

describe('Knuckle', () => {
  it("should print Knuckle's version", async () => {
    let response = await cmd.run('node', [binScriptPath, '-v']);
    expect(response.trim()).toBe(packageJson.version);

    response = await cmd.run('node', [binScriptPath, '--version']);
    expect(response.trim()).toBe(packageJson.version);
  });
});

describe('Usage', () => {
  it('should print the usage', async () => {
    let response = await cmd.run('node', [binScriptPath]);
    expect(response).toMatchSnapshot();

    response = await cmd.run('node', [binScriptPath, '-h']);
    expect(response).toMatchSnapshot();

    response = await cmd.run('node', [binScriptPath, '--help']);
    expect(response).toMatchSnapshot();
  });
});

describe('Error', () => {
  it('should print an `unknown option` error', async () => {
    try {
      await cmd.run('node', [binScriptPath, '--bad-option']);
    } catch (err) {
      expect(err.trim()).toMatchSnapshot();
    }
  });
});
