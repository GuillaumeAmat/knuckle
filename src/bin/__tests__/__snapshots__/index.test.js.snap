// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should print an \`unknown option\` error 1`] = `"error: unknown option '--bad-option'"`;

exports[`Usage should print the usage 1`] = `
"Usage: index <command> [options]

Options:
  -v, --version                    output the version number
  -h, --help                       output usage information

Commands:
  add [toolname...]                Ask Knuckle to handle new tools
  eslint [options] [path...]       An AST-based pattern checker for JavaScript
  lint-staged [options] [path...]  Lint files staged by git
  prettier [options] [path...]     Prettier is an opinionated code formatter
  tslint [options] [path...]       An extensible linter for the TypeScript language
  remove <toolname> [toolname...]  Tell to Knuckle to not handle those tools anymore
  up [options]                     Create the configuration files
  commitlint [options] [path...]   Lint your conventional commits
"
`;

exports[`Usage should print the usage 2`] = `
"Usage: index <command> [options]

Options:
  -v, --version                    output the version number
  -h, --help                       output usage information

Commands:
  add [toolname...]                Ask Knuckle to handle new tools
  eslint [options] [path...]       An AST-based pattern checker for JavaScript
  lint-staged [options] [path...]  Lint files staged by git
  prettier [options] [path...]     Prettier is an opinionated code formatter
  tslint [options] [path...]       An extensible linter for the TypeScript language
  remove <toolname> [toolname...]  Tell to Knuckle to not handle those tools anymore
  up [options]                     Create the configuration files
  commitlint [options] [path...]   Lint your conventional commits
"
`;

exports[`Usage should print the usage 3`] = `
"Usage: index <command> [options]

Options:
  -v, --version                    output the version number
  -h, --help                       output usage information

Commands:
  add [toolname...]                Ask Knuckle to handle new tools
  eslint [options] [path...]       An AST-based pattern checker for JavaScript
  lint-staged [options] [path...]  Lint files staged by git
  prettier [options] [path...]     Prettier is an opinionated code formatter
  tslint [options] [path...]       An extensible linter for the TypeScript language
  remove <toolname> [toolname...]  Tell to Knuckle to not handle those tools anymore
  up [options]                     Create the configuration files
  commitlint [options] [path...]   Lint your conventional commits
"
`;
